### PR TITLE
Complete UI and logic account registration

### PR DESCRIPTION
**Pull request by nguyentrieutien1**
Complete UI and logic account registration

### CODE REVIEW

**Comment by nguyentrieutien1** [comment]
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/73273?issue_count=34&issue_position=11&next_issue_id=73272)

## WHAT (optional)
- Create UI and implement the logic to account registration

## HOW
- using the bootstrap to create the UI
- using custom validates to custom the method for password
- using has_secure_password to hash the password
- using mailter to send email when user sign up

## WHY (optional)

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      flash[:info] = t("email.activation.subject")
+      redirect_to root_url, status: :see_other
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def user_params
+    params.permit(:email, :first_name, :last_name, :password, :password_confirmation)
+  end
```
Comment (manhhnee) [inline app/controllers/registrations_controller.rb:21]: missing phone_number field

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,58 @@
...
 
 class User < ApplicationRecord
+  has_secure_password
+  
+  validates :email, presence: true
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :password, presence: true
+  validates_confirmation_of :password_digest
+  validate :password_requirements_are_met
```
Comment (manhhnee) [inline app/models/user.rb:11]: missing phone_number validates

File: app/views/static_pages/home.html.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<h1>This is home</h1>
```
Comment (manhhnee) [inline app/views/static_pages/home.html.erb:1]: I18n

**Review by manhhnee — COMMENTED** [review]
commented

**Comment by nguyentrieutien1** [comment]
ready
image
image

**Comment by nguyentrieutien1** [comment]
ready

Related Tickets
[#TicketID](https://edu-redmine.sun-asterisk.vn/issues/73273?issue_count=34&issue_position=11&next_issue_id=73272)
WHAT (optional)
Create UI and implement the logic to account registration
HOW
- using the bootstrap to create the UI
- using custom validates to custom the method for password
- using has_secure_password to hash the password
- using mailter to send email when user sign up
WHY (optional)

**Comment by nguyentrieutien1** [comment]
ready
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/73273?issue_count=34&issue_position=11&next_issue_id=73272

## WHAT (optional)
- Create UI and implement the logic to account registration
## HOW
- using the bootstrap to create the UI.
- using custom validates to custom the method for password.
- using has_secure_password to hash the password
- using mailter to send email when user sign up

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

File: Gemfile [hunk 4]
```diff
@@ -24,8 +24,17 @@ gem "figaro"
 
 gem "mysql2"
 
+gem 'htmlbeautifier'
+
+gem 'bootstrap', '~> 4.0'
```
Comment (vanvtt-0952) [inline Gemfile:29]: Nháy đôi

File: app/assets/stylesheets/style/main.css [hunk 5]
```diff
@@ -0,0 +1,49 @@
+/* CSS Reset */
+
+* {
+    margin: 0;
```
Comment (vanvtt-0952) [inline app/assets/stylesheets/style/main.css:4]: Sai indent - 2 space nhá, ko phải tab

File: app/controllers/account_activations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class AccountActivationsController < ApplicationController
```
Comment (vanvtt-0952) [inline app/controllers/account_activations_controller.rb:3]: Ủa sao lại là controller, nếu đã controller thì nó là accounts chứ thêm activations chi thế

File: app/controllers/account_activations_controller.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    handle_error_and_redirect do
+      return if @user.authenticated?(:activation, params[:activation_digest])
+    end
+  end
+
+  def load_user
+    @user = User.find_by activation_digest: params[:activation_digest]
+    return if @user
+
+    redirect_to root_path
```
Comment (vanvtt-0952) [inline app/controllers/account_activations_controller.rb:30]: thêm message để thân thiện người dùng

File: app/controllers/registrations_controller.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    if @user.save
+      @user.send_activation_email
+      flash[:info] = t("email.activation.subject")
+      redirect_to root_url, status: :see_other
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def user_params
```
Comment (vanvtt-0952) [inline app/controllers/registrations_controller.rb:19]: private cho method này

**Comment by nguyentrieutien1** [comment]
ready

**Pull request merged**